### PR TITLE
build: only push docker images in the release build

### LINKED ISSUE #16267: build: non-release builds are getting pushed to docker hub
Something about our detection of which builds qualify as official releases isn't working properly, since it looks like CI builds against the release-1.0 branch are getting considered official releases and pushed to Docker Hub:

screen shot 2017-06-01 at 10 51 15 am

In the screenshot above, `release-1.0` shouldn't exist and `latest` should have been pushed at the same time as `v1.0.1`.

### PR DESCRIPTION
**Pull request by tamird**
This introduces some coupling to TeamCity, but this stuff is already
quite tightly coupled.

Closes #16267.

### CODE REVIEW

File: build/teamcity-publish-artifacts.sh [hunk 1]
```diff
@@ -9,7 +9,7 @@ export BUILDER_HIDE_GOPATH_SRC=1
 
 build/teamcity-publish-s3-binaries.sh "$@"
 
-if [ "$TC_BUILD_BRANCH" != master ]; then
+if [ "$TEAMCITY_BUILDCONF_NAME" == 'Publish Releases' ]; then
```
Comment (a-robinson) [inline build/teamcity-publish-artifacts.sh]: Won't this still trigger for alpha releases, judging by https://teamcity.cockroachdb.com/viewType.html?buildTypeId=Cockroach_ReleaseBuild?

Comment (tamird) [inline build/teamcity-publish-artifacts.sh]: Indeed. Updated to ignore alphas.

File: build/teamcity-publish-artifacts.sh [hunk 2]
```diff
@@ -9,7 +9,7 @@ export BUILDER_HIDE_GOPATH_SRC=1
 
 build/teamcity-publish-s3-binaries.sh "$@"
 
-if [ "$TC_BUILD_BRANCH" != master ]; then
+if [ "$TC_BUILD_BRANCH" != *alpha* ] && [ "$TEAMCITY_BUILDCONF_NAME" == 'Publish Releases' ]; then
```
Comment (a-robinson) [inline build/teamcity-publish-artifacts.sh]: I believe the first comparison needs double braces to work properly, i.e. `[[ "$TC_BUILD_BRANCH" != *alpha* ]]`

**Review by a-robinson — APPROVED** [review]
LGTM other than the one comment

File: build/teamcity-publish-artifacts.sh (hunk 2 above)
Comment (tamird) [inline build/teamcity-publish-artifacts.sh]: Ugh, nice catch!

**Pull request merged**